scripts/pre-push

```diff
@@ -1,13 +1,89 @@
 #!/bin/bash
 
-echo "Running unit tests..."
+set -e  # Exit on any error
 
-./gradlew clean # First clean the project
-./gradlew test # Run the tests
+echo "🚀 Running pre-push checks..."
 
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
 
-status=$?
+# Function to print colored output
+print_status() {
+    echo -e "${BLUE}[INFO]${NC} $1"
+}
 
-# return 1 exit code if running checks fails
-[ $status -ne 0 ] && exit 1
-exit 0
+print_success() {
+    echo -e "${GREEN}[SUCCESS]${NC} $1"
+}
+
+print_error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+}
+
+print_warning() {
+    echo -e "${YELLOW}[WARNING]${NC} $1"
+}
+
+# 1. Check branch name
+print_status "Validating branch name..."
+LC_ALL=C
+local_branch="$(git rev-parse --abbrev-ref HEAD)"
+valid_branch_regex="^(feat|fix|hotfix|refactor|release|docs|chore|test|ci)\/[a-z0-9._-]+$"
+message="There is something wrong with your branch name. Branch names in this project must adhere to this contract: $valid_branch_regex. Your push will be rejected. You should rename your branch to a valid name and try again."
+
+if [[ ! $local_branch =~ $valid_branch_regex ]]
+then
+    print_error "$message"
+    exit 1
+fi
+print_success "Branch name validation passed"
+
+# 2. Clean the project
+print_status "Cleaning project..."
+./gradlew clean
+print_success "Project cleaned"
+
+# 3. Run unit tests
+print_status "Running unit tests..."
+if ! ./gradlew test; then
+    print_error "Unit tests failed. Please fix failing tests before pushing."
+    exit 1
+fi
+print_success "All unit tests passed"
+
+# 4. Build all modules
+print_status "Building all modules..."
+build_failed=false
+
+# Build core module (Kotlin/JVM)
+if ! ./gradlew :core:assemble; then
+    print_error "Build failed for core module"
+    build_failed=true
+fi
+
+# Build android module
+if ! ./gradlew :android:assembleRelease; then
+    print_error "Build failed for android module"
+    build_failed=true
+fi
+
+# Build integration modules
+for integration in adjust appsflyer firebase facebook braze; do
+    if ! ./gradlew ":integrations:$integration:assembleRelease"; then
+        print_error "Build failed for integrations:$integration module"
+        build_failed=true
+    fi
+done
+
+if [ "$build_failed" = true ]; then
+    print_error "Build failed for one or more modules. Please fix build issues before pushing."
+    exit 1
+fi
+print_success "All modules built successfully"
```

Comment:
This might not be easy to maintain. Should we consider simply running `./gradlew build` command, which should build all the available modules? Otherwise, in future if we plan to add new modules, we need to clean up the local `.git/hooks` directory.